packages/pyroscope-flamegraph/src/format/format.ts

```diff
@@ -90,6 +91,73 @@ class DurationFormatter {
   }
 }
 
+// this is a class and not a function because we can save some time by
+//   precalculating divider and suffix and not doing it on each iteration
+class SubSecondDurationFormatter {
+  divider = 1;
+
+  suffix = 'second';
+
+  durations: [number, string][] = [
+    [60, 'minute'],
+    [60, 'hour'],
+    [24, 'day'],
+    [30, 'month'],
+    [12, 'year'],
+  ];
+
+  subSecondDurations: [number, string][] = [
+    [1000, 'ms'],
+    [1000, 'μs'],
+  ];
+
+  units = '';
+
+  constructor(maxDur: number, units?: string) {
```

Comment:
`units` isn't really used by any of the users, right? Therefore I'd remove it altogether.